153_save_sensor_data/sensor_data_store.py

```diff
@@ -0,0 +1,248 @@
+
+# The measurement interval can be changed with this, but it still seems to take 
+# measurements at random time intervals just can be set "close" to this 
+# interval. The fastest interval combined with interpolation or just most recent
+# reading seems like the best way to me.
+# Can be increased to make a recent reading closer to another time step if 
+# we tried to synchronize it. Interpolation can be done either before data is 
+# saved or after data is retrieved from storage but prior to packaging in JSON.
+# scd.measurement_interval = INTERVAL 
+
+# Function Definitions
+
+def initialize_sensor():
+    ''' Function to initialize the sensor.
+        From Carter's Script at Adafruit, it is recommended to start with a 
+        frequency of 50 Khz due to "Clock Stretching". '''
+    i2c = busio.I2C(board.SCL, board.SDA, frequency=50000)
+    scd = adafruit_scd30.SCD30(i2c)
+    return scd
+ 
+def make_file_paths():
+    ''' Create the directories to store the saved data '''
+    Path("scd_data/raw").mkdir(parents=True,exist_ok=True)
+    Path("scd_data/interpolated").mkdir(exist_ok=True)
+
+
+def interpolate_linear(time_of_interest, data_set_low, data_set_high):
+    ''' # This function performs an interpolation to a time of interest
+        # given two data points outside of the time of interest using
+        # a linear method. '''
+    given_time2 = data_set_high['seconds']
+    given_time1 = data_set_low['seconds']
+    temp_high, temp_low = data_set_high['temp'], data_set_low['temp']
+    humidity_high = data_set_high['humidity']
+    humidity_low = data_set_low['humidity']
+    co2_high, co2_low = data_set_high['co2'], data_set_low['co2']
+    # Difference between the time of two measurements
+    time_difference = given_time2 - given_time1
+    #First determine the slope of the path between the two times.
+    # This is the slope of the line known as a linear interpolant.
+    # The slope of a line where the horizontal axis is time,
+    # and the vertical axis is the value to interpolate, is the
+    # rate of change in the value per unit time.
+    change_rate_temp = (temp_high - temp_low) / time_difference
+    change_rate_humidity = (humidity_high - humidity_low) / time_difference
+    change_rate_c02 = (co2_high - co2_low) / time_difference
+
+    # Difference between the time of interest and the fistTime
+    time_difference = time_of_interest - given_time1
+    # The interpolated data value is the early value plus the time difference
+    # multiplied by the slope (change rate)
+    interpolated_temp = temp_low + change_rate_temp*time_difference
+    interpolated_humidity = humidity_low + change_rate_humidity*time_difference
+    interpolated_co2 = co2_low + change_rate_c02*time_difference
+    # The interpolated value is returned out of this function.
+    return dict(seconds=time_of_interest, co2=interpolated_co2,
+               temp=interpolated_temp,
+               humidity=interpolated_humidity)
```

Comment:
```suggestion
    return dict(seconds=time_of_interest, co2=interpolated_co2,
                temp=interpolated_temp,
                humidity=interpolated_humidity)
```
The args should be aligned (`temp` under `seconds`).